src/openms_gui/source/VISUAL/DIALOGS/ToolsDialog.cpp

```diff
@@ -174,36 +167,65 @@ namespace OpenMS
 
   }
 
-  void ToolsDialog::createINI_()
+  Param ToolsDialog::getParamFromIni_(const String& toolName)
   {
     QStringList args{ "-write_ini", ini_file_.toQString(), "-log", (ini_file_+".log").toQString() };
     QProcess qp;
-    String executable = File::findSiblingTOPPExecutable(getTool());
+    String executable = File::findSiblingTOPPExecutable(toolName);
     qp.start(executable.toQString(), args);
     const bool success = qp.waitForFinished(-1); // wait till job is finished
     if (qp.error() == QProcess::FailedToStart || success == false || qp.exitStatus() != 0 || qp.exitCode() != 0)
     {
-      QMessageBox::critical(this, "Error", (String("Could not execute '") + executable + "'!\n\nMake sure the TOPP tools are present in '" + File::getExecutablePath() + "',  that you have permission to write to the temporary file path, and that there is space left in the temporary file path.").c_str());
-      return;
+        QMessageBox::critical(this, "Error", (String("Could not execute '") + executable + "'!\n\nMake sure the TOPP tools are present in '" + File::getExecutablePath() + "',  that you have permission to write to the temporary file path, and that there is space left in the temporary file path.").c_str());
+        // TODO handle error
     }
     else if (!File::exists(ini_file_))
     {
-      QMessageBox::critical(this, "Error", (String("Could find requested INI file '") + ini_file_ + "'!").c_str());
-      return;
+        QMessageBox::critical(this, "Error", (String("Could find requested INI file '") + ini_file_ + "'!").c_str());
+        // TODO handle error
+    }
+    Param toolP;
+    ParamXMLFile paramFile;
+    paramFile.load((ini_file_).c_str(), toolP);
+
+    return toolP;
+  }
+
+  std::vector<LayerData::DataType> ToolsDialog::getTypesFromParam_(const Param& p) const
+  {
+    // Containing all types a tool is compatible with
+    std::vector<LayerData::DataType> types;
+    for (const auto& entry : p)
+    {
+      if (entry.name == "in")
+      {
+        // Map all file extension to a LayerData::DataType
+        for (auto& fileExtension : entry.valid_strings)
+        {
+          const auto& iter = tool_map_.find(fileExtension);
```

Comment:
I changed to map the reflect your suggestion. I had to change the following lines:
```suggestion
          // valid string is of format "*.TYPE" -> convert to substr "TYPE" in order to be comparable.
          const String& fileTypeStr = fileExtension.substr(2, fileExtension.size());
          const auto& iter = tool_map_.find(FileTypes::nameToType(subStr));
```
Each entry in of `valid_strings` is of format "*.TYPE". I sliced of the first two characters in order to be able to convert it to a `FileTypes::Type` with `FileTypes::nametoType` so it can be mapped. Is this what you had in mind? I don't see a way to get completely rid of using strings to determine the type.